features/FEATURE_BLE/source/generic/GenericGap.cpp

```diff
@@ -1084,7 +1231,17 @@ void GenericGap::on_connection_complete(const pal::GapConnectionCompleteEvent& e
             address.data(),
             &connection_params
         );
-    } else {
+
+        // Now starts pairing or authentication procedures if required
+        if(needs_pairing) {
+            SecurityManager &sm = createBLEInstance()->getSecurityManager();
+            sm.requestPairing(e.connection_handle);
+        }
+        else if(needs_authentication) {
+            SecurityManager &sm = createBLEInstance()->getSecurityManager();
+            sm.requestAuthentication(e.connection_handle);
```

Comment:
GAP Authentication is different from security manager authentication 🙄 .
We do not have all the mechanisms in place to handle this; we need a global flag in the DB.
I would drop this and let it empty.

Reply:
Haha never assume anything with the Bluetooth spec 😆 
Thanks for pointing that out 😄

Reply:
there is a flag for that:
authenticated in the ControlBlock_t of the Security Manager